Sources/WebSocket.swift

```diff
@@ -710,11 +710,11 @@ open class WebSocket : NSObject, StreamDelegate, WebSocketClient, WSStreamDelega
      Disconnect the stream object and notifies the delegate.
      */
     private func disconnectStream(_ error: Error?, runDelegate: Bool = true) {
-        if error == nil {
-            writeQueue.waitUntilAllOperationsAreFinished()
-        } else {
+        if error != nil {
             writeQueue.cancelAllOperations()
```

Comment:
cancelAllOperations DOES NOT stop currently executing blocks from finishing. We must wait until those have finished before we set `outputStream` to nil as the blocks are attempting to access `outputStream`.